ui/src/App.tsx

```diff
@@ -1,11 +1,25 @@
 import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import Header from './Header';
+import Routes from './Routes';
+
+import firebase from './utils/firebase';
+
+const App: React.FC = () => {
+  const [currentUser, setCurrentUser] = React.useState<firebase.User | null>(null);
+
+  React.useEffect(() => {
+    const unsubscribe = firebase.auth().onAuthStateChanged((u) => setCurrentUser(u));
```

Comment:
Same here, not sure if it make sense to have this so high in the hierarchy, but it makes handing sign in from the sign in page and sign out from the header quite easy 🤔

Reply:
If all of your pages require this state handling then it makes sense for now. You can also combine the user state with this effect in a custom hook. They rely to each other so it makes sense.

That way you can also easily move the functionality when it doesn’t make sense to be here anymore.